File: cpp/pangram/pangram.cpp
#include <ctype.h>
#include <unordered_set>
#include "pangram.h"

namespace pangram {

    bool is_pangram(std::string const &input) {

        std::unordered_set<char> letters_in_input;
        for (char const letter: input) {
            if (isalpha(letter)) {
                letters_in_input.insert(tolower(letter));
            }
        }
        return letters_in_input.size() == 26;
    }

}  // namespace pangram

File: haskell/pangram/src/Pangram.hs
module Pangram (isPangram) where

import qualified Data.Char as Char
import qualified Data.HashSet as HashSet
import           Data.HashSet (HashSet)

isPangram :: String -> Bool
isPangram text = all (\letters -> letters `elem` map Char.toLower text) lowerAlpha

lowerAlpha :: HashSet Char
lowerAlpha = HashSet.fromList ['a'..'z']
